### PR TITLE
[CI] Add initial inductor cpu smoketest for performance

### PR DESCRIPTION
**Pull request by zxd1997066**
cc @jgong5 @mingfeima @XiaobingSuper @sanchitintel @ashokei @jingxu10 @voznesenskym @penguinwu @EikanWang @Guobing-Chen @zhuhaozhe @blzheng @wenzhe-nrv @jiayisunx @chenyang78 @aakhundov @kadeng

### CODE REVIEW

**Comment by chuanqi129** [comment]
Have tested 3 rounds based on linux.12xlarge runner, the performance looks stable for ci test check. And define the speedup ratio target as 0.93*speedup_avg.
Round1: https://github.com/pytorch/pytorch/actions/runs/7626013836/job/20772162550?pr=116456
Round2: https://github.com/pytorch/pytorch/actions/runs/7627037296/job/20775557751?pr=116456
Round3: https://github.com/pytorch/pytorch/actions/runs/7627806742/job/20778138286?pr=116456

#name | data_type | shape | wrapper | perf_speedup_target_c5_12xlarge | Round1 speedup | Round1 abs_latency | Round2 speedup | Round2 abs_latency | Round3 speedup | Round3 abs_latency | speedup_avg
-- | -- | -- | -- | -- | -- | -- | -- | -- | -- | -- | --
phlippe_densenet | float32 | static | default | 1.99590617 | 2.127598 | 33.888912 | 2.145434 | 33.439566 | 2.165375 | 32.1873 | 2.14613567
basic_gnn_gcn | float32 | dynamic | default | 1.24639561 | 1.338088 | 14.180395 | 1.340817 | 14.17623 | 1.341726 | 13.830302 | 1.34021033
llama_v2_7b_16h | float32 | dynamic | default | 1.27455818 | 1.355591 | 1268.4725 | 1.406341 | 1201.1005 | 1.349546 | 1251.2688 | 1.37049267
resnet50 | float32 | dynamic | default | 2.28794694 | 2.46323 | 96.464452 | 2.464778 | 96.152014 | 2.452466 | 95.004308 | 2.460158
timm_efficientnet | float32 | static | cpp | 2.72195686 | 2.879213 | 79.60927 | 2.965314 | 76.654795 | 2.935979 | 76.030633 | 2.92683533
mobilenet_v3_large | float32 | static | cpp | 3.02274304 | 3.256662 | 18.767327 | 3.249648 | 18.710148 | 3.244474 | 17.979417 | 3.25026133
timm_resnest | float32 | dynamic | cpp | 2.10118744 | 2.261502 | 78.796368 | 2.260884 | 78.594948 | 2.255638 | 77.973561 | 2.25934133
shufflenet_v2_x1_0 | float32 | dynamic | cpp | 1.8976929 | 2.045257 | 30.682947 | 2.054571 | 30.494747 | 2.021762 | 29.605321 | 2.04053
hf_GPT2 | float32 | dynamic | cpp | 1.1183002 | 1.21309 | 47.609004 | 1.221014 | 47.297136 | 1.173316 | 46.889306 | 1.20247333

File: benchmarks/dynamo/expected_ci_speedup_inductor_torchbench_cpu.csv [hunk 1]
```diff
@@ -0,0 +1,12 @@
...
+phlippe_densenet,float32,static,default,1.99590617
+basic_gnn_gcn,float32,dynamic,default,1.24639561
+llama_v2_7b_16h,float32,dynamic,default,1.27455818
+resnet50,float32,dynamic,default,2.28794694
+timm_efficientnet,float32,static,cpp,2.72195686
+mobilenet_v3_large,float32,static,cpp,3.02274304
+timm_resnest,float32,dynamic,cpp,2.10118744
+shufflenet_v2_x1_0,float32,dynamic,cpp,1.8976929
+#hf_GPT2,float32,dynamic,cpp,1.6702305
+hf_GPT2,float32,dynamic,cpp,1.1183002
```
Comment (jgong5) [inline benchmarks/dynamo/expected_ci_speedup_inductor_torchbench_cpu.csv:12]: Which one is preferred?

File: benchmarks/dynamo/expected_ci_speedup_inductor_torchbench_cpu.csv [hunk 2]
```diff
@@ -0,0 +1,12 @@
+#name,data_type,shape,wrapper,perf_speedup_target_c5_12xlarge
+#timm_vision_transformer,float32,static,default,1.1585628
```
Comment (jgong5) [inline benchmarks/dynamo/expected_ci_speedup_inductor_torchbench_cpu.csv:2]: Can you explain why we comment out this line?

File: .ci/pytorch/test.sh [hunk 3]
```diff
@@ -521,6 +521,51 @@ test_inductor_torchbench_smoketest_perf() {
   done
 }
 
+test_inductor_torchbench_cpu_smoketest_perf(){
+  TEST_REPORTS_DIR=$(pwd)/test/test-reports
+  mkdir -p "$TEST_REPORTS_DIR"
+
+  #set jemalloc
+  #JEMALLOC_LIB="$(dirname "$(which python)")/../lib/libjemalloc.so"
+  JEMALLOC_LIB="/usr/lib/x86_64-linux-gnu/libjemalloc.so.2"
```
Comment (jgong5) [inline .ci/pytorch/test.sh:530]: How do we make sure libjemalloc is installed?

File: .ci/pytorch/test.sh [hunk 4]
```diff
@@ -521,6 +521,51 @@ test_inductor_torchbench_smoketest_perf() {
...
+  IOMP_LIB="$(dirname "$(which python)")/../lib/libiomp5.so"
```
Comment (jgong5) [inline .ci/pytorch/test.sh:531]: ditto: how do we make sure libiomp is installed?

Comment (chuanqi129) [reply .ci/pytorch/test.sh:531]: The base docker image build need install `mkl` conda package, and it depends on `intel-openmp` conda package. Refer: https://github.com/pytorch/pytorch/blob/main/.ci/docker/common/install_conda.sh#L54

File: .ci/pytorch/test.sh (hunk 3 above)
Comment (chuanqi129) [reply .ci/pytorch/test.sh:530]: It's installed in base docker build stage, refer https://github.com/pytorch/pytorch/blob/main/.ci/docker/common/install_base.sh#L64

File: benchmarks/dynamo/expected_ci_speedup_inductor_torchbench_cpu.csv (hunk 1 above)
Comment (jgong5) [reply benchmarks/dynamo/expected_ci_speedup_inductor_torchbench_cpu.csv:12]: Can we add a comment to explain the situation?

**Review by jgong5 — APPROVED** [review]

**Comment by chuanqi129** [comment]
Hi @atalman @huydhn , could you please also help to review this inductor cpu performance smoke test enabling in CI?

File: .ci/pytorch/test.sh [hunk 5]
```diff
@@ -524,6 +524,50 @@ test_inductor_torchbench_smoketest_perf() {
...
+      taskset -c 0-"$end_core" python benchmarks/dynamo/torchbench.py \
+        --inference --performance --"$data_type" -dcpu -n50 --only "$model_name" --dynamic-shapes \
+        --dynamic-batch-only --freezing --timeout 9000 --backend=inductor --output "$output_name"
+    else
+      taskset -c 0-"$end_core" python benchmarks/dynamo/torchbench.py \
+        --inference --performance --"$data_type" -dcpu -n50 --only "$model_name" \
+        --freezing --timeout 9000 --backend=inductor --output "$output_name"
+    fi
+    cat "$output_name"
+    # The threshold value needs to be actively maintained to make this check useful.
```
Comment (atalman) [inline .ci/pytorch/test.sh:566]: Hi @chuanqi129 it is a plan for Intel to maintain this check ?

**Review by atalman — APPROVED** [review]

**Pull request closed without merging**